test/jsonapi/view_test.exs

```diff
@@ -33,6 +33,26 @@ defmodule JSONAPI.ViewTest do
     assert PostView.url_for_rel(%{id: 1}, "comments", %Plug.Conn{}) == "http://www.example.com/api/posts/1/relationships/comments"
   end
 
+  test "url_for/2 with configured host" do
+    Application.put_env(:jsonapi, :host, "www.otherhost.com")
+    assert PostView.url_for([], %Plug.Conn{}) == "http://www.otherhost.com/api/posts"
+    assert PostView.url_for(%{id: 1}, %Plug.Conn{}) == "http://www.otherhost.com/api/posts/1"
+    assert PostView.url_for_rel([], "comments", %Plug.Conn{}) == "http://www.otherhost.com/api/posts/relationships/comments"
+    assert PostView.url_for_rel(%{id: 1}, "comments", %Plug.Conn{}) == "http://www.otherhost.com/api/posts/1/relationships/comments"
+    Application.delete_env(:jsonapi, :host)
+  end
+
+  test "url_for/2 with configured scheme" do
+    Application.put_env(:jsonapi, :host, "otherhost.com")
```

Comment:
If you take a peek at my branch you'll see I consolidated these tests together for this very reason.  Unfortunately since the tests run currently, this can still cause an issue just less frequently.  Can we clean this up a bit?